fix: fix generation of NPM release package (#75, #78)

aeworxet opened the pull request:
This PR adds building of the TypeScript code (transpilation to JavaScript) during release process, which was not required before migration of the codebase to TS, and therefore did not exist.
Due to absence of this step, after migration to TS NPM package was released without `lib` directory, which made the package unusable for end users.

Besides adding the building step, this PR also modifies `files` array of the `package.json` to include in the package file `API.md`.


Fixes https://github.com/asyncapi/bundler/issues/75
Fixes https://github.com/asyncapi/bundler/issues/78

Souvikns commented on package.json:
```diff
@@ -34,7 +35,7 @@
     "LICENSE",
     "README.md",
-    "/lib"
+    "lib/"
```
```suggestion
    "/lib"
```

I am confused is it `/lib` or `lib/`?
https://github.com/asyncapi/bundler/pull/76/files changed it to `/lib`, and also cli uses https://github.com/asyncapi/cli/blob/bf66a0c58362c2f8012e5285b6f4bf8ace26311d/package.json#L84 so lets just leave it as `/lib`

magicmatatjahu replied:
You can run `npm run build && npm pack` and check if `/lib` folder is inside packed package :)

magicmatatjahu replied:
> /lib and lib/ works, so the only concern is consistency really

I remember that some version in the parser v2 did not work 😅 so we should check that by scripts `npm run build && npm pack`.

aeworxet commented:
Changed `"lib/"` to `"/lib"`.

Implemented separation of concerns, as
```
"build": "tsc",
"generate:assets": "npm run docs && npm run generate:readme:toc",
"prepublishOnly": "npm run build && npm run generate:assets"
```

Modified `files` array to include `API.md`.
```
"files": [
  "API.md",
  ...
],
```

Souvikns reviewed (dismissed):
👍🏼 looking good.

aeworxet commented:
Now it's looking even better after returning everything back.

derberg reviewed (approved):
:shipit:

Souvikns reviewed (approved):
👍🏼
